src/CachePlugin.php

```diff
@@ -86,8 +113,16 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
                     $response = $response->withBody($this->streamFactory->createStream($body));
                 }
 
-                $cacheItem->set(['response' => $response, 'body' => $body])
-                    ->expiresAfter($this->getMaxAge($response));
+                $expiresAt = time() + $this->getMaxAge($response);
+                $cacheItem
+                    ->expiresAfter($this->config['cache_lifetime'] + $expiresAt)
+                    ->set([
+                    'response' => $response,
+                    'body' => $body,
+                    'expiresAt' => $expiresAt,
```

Comment:
Yes but that we would like to be able to ask the server if the response is still valid after it has expired. If so, we need to store the response for a longer time that it's max age.